### ISSUE TITLE
Enforce user ability to delete self-owned posts

### ISSUE THREAD

**Issue by icelynjennings** [Jaxboards/Jaxboards#49]
IIRC admins can currently disallow deletion or editing of a user's own posts, which poses as a potential privacy issue. To protect users from harassment, the permission to edit and delete owned posts should be enforced across all groups of users.

I suggest a platform-wide database edit for that permission across all forums and further disabling the API endpoint for changing the permission, deprecating the db column.

**Comment by wtl420** [comment]
I agree that giving administrators total control over user-submitted content seems misguided.

I think it's OK that edit posts feature can be disabled with the way editing works right now. Editing posts is kind of weird in that they could easily be used for good and evil and I think it should be up to board administrators to decide what to do there. Maybe in the future we can have some sort of edit history feature? Though I'm not sure how to make it work in a way that gives the original poster the most power over it w/o it being abused.

I've given a lot of thought to post deletion and I agree that users should always be able to remove their content, from their posts to their accounts themselves. I think there's a good way to balance this without it causing too much trouble- MyBB has a feature called 'soft deletion', which when a post is deleted, forum staff can still view the post. I like this idea more than the 'trash can' approach we currently have as it keeps the context of deleted content. However, I'd like to take it a step further and have some kind of auto-purge for soft-deleted content- maybe 30 days by default or so, that could be set to 0 to just have instant hard deletion. Soft-deletion could be done in a way with accounts as well to have it work kind of like Twitter deactivations- a user can restore their account or post within the grace period. This doesn't give a user perfect control over their content but I think it's a fair balance.

Of course the problem here is there isn't too much mitigation against malicious forum staff. I'm not sure there's ever going to be a perfect solution against that though- no matter what we do to the software, someone could always be logging content or even restore a backup to maliciously restore user-deleted content.

Quoting posts might always be a problem for post deletion though. Maybe we can do some kind of post embedding like Twitter can do for "quote tweets"? And when a post is deleted the embed would just be like 'this post has been deleted'. Though I like to partially quote posts a lot, I'm not sure how to work that into this kind of embed system. Maybe the embed works like the current quote system but double checks that the text in the tag is actually in the source post? An interesting problem to solve.

**Comment by DevopsIce** [comment]
Good points on total removal of presence. The edge cases you mentioned could be tackled in a future issue, meanwhile the ability to self-delete posts as understood by the db model currently can be an easy win for now.
